src/sdam/monitor.ts

```diff
@@ -352,37 +354,45 @@ function checkServer(monitor: Monitor, callback: Callback<Document | null>) {
   }
 
   // connecting does an implicit `hello`
-  connect(monitor.connectOptions, (err, conn) => {
-    if (err) {
-      monitor.connection = null;
-
-      awaited = false;
-      onHeartbeatFailed(err);
-      return;
+  (async () => {
+    const socket = await makeSocket(monitor.connectOptions);
+    const connection = makeConnection(monitor.connectOptions, socket);
+    start = now();
```

Comment:
Maybe it would be nice to add a comment here and on the 338 line (with another `start = now()`) explaining what you were telling me about when we expect it to be called.

Reply:
Added doc comments